src/transactions/TransactionFrame.cpp

```diff
@@ -1722,8 +1719,18 @@ TransactionFrame::apply(Application& app, AbstractLedgerTxn& ltx,
         //  when applying, a failure during tx validation means that
         //  we'll skip trying to apply operations but we'll still
         //  process the sequence number if needed
-        auto cv =
-            commonValid(app, signatureChecker, ltxTx, 0, true, chargeFee, 0, 0);
+        std::optional<FeePair> sorobanResourceFee;
+        if (protocolVersionStartsFrom(ledgerVersion,
+                                      SOROBAN_PROTOCOL_VERSION) &&
+            isSoroban())
```

Comment:
Yes, I meant not returning 0 ever as it can cause a divergence if the fee isn't recomputed (the latest update removed 0 fees, so it's all good).

>I've thought a bit about that and I'm honestly not sure about this whole block of validations that assumes that somehow we might end up applying an invalid tx (I guess SCP doesn't prohibit that directly?). In any case, this check is consistent with the validations here (which apparently allow a potentially invalid/malformed tx).

interesting, indeed this seems to be the case wrt to other validity checks, so it's fine to keep the version check